### DISCUSSION TITLE
Question about record visibility

### DISCUSSION THREAD

**Discussion by fps37** [jamessimone/apex-rollup#138] [General]
We have a POC using Rollups - but have run into an issue where it seems that the runtime does not have full visibility to records with different owners, despite "without sharing" being active.  Any help would be appreciated.

**Comment by fps37** [comment]
For clarity - the visibility issue results in incomplete rollups, where some relevant records are not being operated upon.

**Reply by jamessimone** [reply]
@fps37 firstly, thanks for bringing this up! I'm sorry you're experiencing an issue and I hope we can work together to figure out where this is going wrong. What is your entry point for using `Rollup` at present? Are you calling the static methods from Apex? Making use of flow and the invocable methods? What sort of rollup operation(s) are you currently testing out - and are you doing anything special using:

- grandparent rollups
- hierarchical rollups
- rollups with a complicated Calc Item Where Clause

Hopefully these questions will help to further our mutual understanding of why record visibility is coming into play even though it should not. Thanks!

**Reply by fps37** [reply]
Currently testing via CMDT, no grandparent or hierarchical rollups, behavior is the same with or without a where clause.  I'd be happy to share more details via a private channel if that would be helpful.

**Reply by jamessimone** [reply]
That would be great. You can find my contact info at the bottom of the Readme!
